src/util/retry.rs

```diff
@@ -13,18 +13,26 @@ use std::time::Duration;
 /// ```rust
 /// # use std::time::Duration;
 /// # use vss_client::error::VssError;
-/// # use vss_client::util::retry::{ExponentialBackoffRetryPolicy, retry, RetryPolicy};
+/// # use vss_client::util::retry::{ExponentialBackoffRetryPolicy, FilteredRetryPolicy, JitteredRetryPolicy, MaxAttemptsRetryPolicy, MaxTotalDelayRetryPolicy, retry, RetryPolicy};
 /// #
 /// # async fn operation() -> Result<i32, VssError>  {
 /// # 	tokio::time::sleep(Duration::from_millis(10)).await;
 /// # 	Ok(42)
 /// # }
 /// #
-/// let retry_policy = ExponentialBackoffRetryPolicy::new(Duration::from_millis(100))
+/// // Hint: Use concrete type of retry_policy for convenience in function return type.
+/// type VssRetryPolicy = FilteredRetryPolicy<JitteredRetryPolicy<MaxTotalDelayRetryPolicy<MaxAttemptsRetryPolicy<ExponentialBackoffRetryPolicy<VssError>>>>, Box<dyn 'static + Send + Sync + Fn(&VssError) -> bool>>;
+///
+/// let retry_policy: VssRetryPolicy = ExponentialBackoffRetryPolicy::new(Duration::from_millis(100))
 /// 	.with_max_attempts(5)
 /// 	.with_max_total_delay(Duration::from_secs(2))
 /// 	.with_max_jitter(Duration::from_millis(30))
-/// 	.skip_retry_on_error(|e| matches!(e, VssError::InvalidRequestError(..)));
+/// 	.skip_retry_on_error(Box::new(|e: &VssError| {
+/// 				matches!(
+/// 					e,
+/// 					VssError::NoSuchKeyError(..) | VssError::InvalidRequestError(..) | VssError::ConflictError(..)
+/// 				)
+/// 			}) as _);
```

Comment:
Let's keep the original example. And then add a separate example showing when `Box` is actually necessary (i.e., if removing `Box` causes a compilation error). It can be minimal in that it doesn't need to use extra wrappers, too.